### PR TITLE
[Intake] Add initial interactivity

### PR DESCRIPTION
**Pull request by NickHeiner**
1. Fleshing out Redux infrastructure.
1. Making the vet name and id show up in the navigation bar. (Faked out version of #3440.)
1. Simulating a file number search with a loading spinner.
1. Resetting to initial state when the user hits "begin next intake" on the final page.

## Known Issues
~~1. Vet name and id in navigation bar does not look right.~~

### CODE REVIEW

File: client/app/intake/IntakeFrame.jsx [hunk 1]
```diff
@@ -11,20 +12,24 @@ import BeginPage from './pages/begin';
...
+import { PAGE_PATHS, REQUEST_STATE } from './constants';
 
-export default class IntakeFrame extends React.PureComponent {
+class IntakeFrame extends React.PureComponent {
   render() {
     const appName = 'Intake';
 
     const Router = this.props.router || BrowserRouter;
 
+    const topMessage = this.props.fileNumberSearchRequestStatus === REQUEST_STATE.SUCCEEDED ?
```
Comment (shanear) [inline client/app/intake/IntakeFrame.jsx]: had to hunt around to figure out what `REQUEST_STATE` meant. Maybe `SEARCH_REQUEST_STATE` or `SEARCH_RESULT`?

File: client/app/intake/pages/begin.jsx [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
     return <div>
       <h1>Welcome to Caseflow Intake!</h1>
       <p>To begin processing this opt-in request, please enter the Veteran ID below.</p>
-      <SearchBar size="small" onSubmit={this.handleSearchSubmit} />
+      <SearchBar
+        size="small"
+        onSubmit={this.handleSearchSubmit}
+        onChange={this.props.setFileNumberSearch}
+        value={this.props.fileNumberSearchInput}
+        loading={this.props.fileNumberSearchRequestStatus === REQUEST_STATE.IN_PROGRESS}
```
Comment (shanear) [inline client/app/intake/pages/begin.jsx]: nice 💯

File: client/app/intake/redux/actions.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export const setFileNumberSearch = (fileNumber) => ({
+  type: ACTIONS.SET_FILE_NUMBER_SEARCH,
+  payload: {
+    fileNumber
+  }
+});
+
+export const doFileNumberSearch = () => (dispatch) => {
+  return new Promise((resolve) => {
```
Comment (shanear) [inline client/app/intake/redux/actions.js]: yayyy! now I just gotta write this API 😁

File: client/app/intake/IntakeFrame.jsx (hunk 1 above)
Comment (shanear) [reply client/app/intake/IntakeFrame.jsx]: oh now that I'm reading through things, this seems to be some sort of paradigm that you cant represent different types of requests with

File: client/app/intake/redux/reducer.js [hunk 4]
```diff
@@ -1 +1,58 @@
-export default (state = {}) => state;
+import { ACTIONS, REQUEST_STATE } from '../constants';
+import { update } from '../../util/ReducerUtil';
+
+const initialState = {
```
Comment (shanear) [inline client/app/intake/redux/reducer.js]: is there a way for me to plug these in from the ReactOnRails stuff? Some of these may come from the DB if the page is reloaded.

File: client/app/intake/IntakeFrame.jsx (hunk 1 above)
Comment (NickHeiner) [reply client/app/intake/IntakeFrame.jsx]: Yep! It's meant to be generic.

File: client/app/intake/redux/actions.js (hunk 3 above)
Comment (NickHeiner) [reply client/app/intake/redux/actions.js]: image

File: client/app/intake/redux/reducer.js (hunk 4 above)
Comment (NickHeiner) [reply client/app/intake/redux/reducer.js]: Yes. You can pass arbitrary props through via `/app/views/intake/index.html.erb` and I'll be able to pre-load them into the state.

**Pull request merged**